AK/SourceGenerator.h

```diff
@@ -75,10 +82,23 @@ class SourceGenerator {
         }
     }
 
+    void append_include(StringView path)
+    {
+        append(String::formatted("#include <{}>\n", path));
+    }
+
+    SourceGenerator fork_namespace(StringView namespace_)
+    {
+        append(String::formatted("\nnamespace {} {{\n", namespace_));
+        auto suffix = String::formatted("\n}} // namespace {}\n", namespace_);
+        return SourceGenerator { m_builder, m_mapping, m_opening, m_closing, suffix };
```

Comment:
That's a UAF. The SourceGenerator only recieves a StringView, but the string that is referenced is destroyed when the fork_namespace function exits. You probably want to keep the suffix as a string and move() it into the forked generator.

Reply:
I spent a good while debugging that exact bug, when I had the `m_suffix` member as `StringView`.
I guess it's currently working without the move() by luck.

I'm more familiar with Rust's default move, so I'm definitely less familiar with when C++ needs move().
Thanks :+1:

Reply:
The point here is not move(), the point is that with StringView you don't own the string data. In Rust terms: it's a `&str` where the lifetime of the borrowed data has expired.